api/src/main/java/io/opentelemetry/trace/propagation/JaegerPropagator.java

```diff
@@ -0,0 +1,200 @@
+ * href=https://www.jaegertracing.io/docs/client-libraries/#propagation-format>Jaeger Propogation
+ * Format</a>.
+ */
+@Immutable
+public class JaegerPropagator implements HttpTextFormat {
+
+  private static final Logger logger = Logger.getLogger(HttpTraceContext.class.getName());
+
+  static final String TRACE_ID_HEADER = "uber-trace-id";
+  // Parent span has been deprecated but Jaeger propagation protocol requires it
+  static final String DEPRECATED_PARENT_SPAN = "0";
+  static final String SEPARATOR = ":";
+
+  private static final String IS_SAMPLED = "1";
+  private static final String NOT_SAMPLED = "0";
+
+  private static final int MAX_TRACE_ID_LENGTH = 2 * TraceId.getSize();
+  private static final int MAX_SPAN_ID_LENGTH = 2 * SpanId.getSize();
+  private static final int MAX_FLAGS_LENGTH = 2;
+  private static final TraceFlags SAMPLED_FLAGS = TraceFlags.builder().setIsSampled(true).build();
+  private static final TraceFlags NOT_SAMPLED_FLAGS =
+      TraceFlags.builder().setIsSampled(false).build();
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_ID_HEADER);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, C carrier, Setter<C> setter) {
+    checkNotNull(context, "context");
+    checkNotNull(setter, "setter");
+    checkNotNull(carrier, "carrier");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
+    if (span == null) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+    String sampled = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+
+    setter.set(
+        carrier,
+        TRACE_ID_HEADER,
+        spanContext.getTraceId().toLowerBase16()
+            + SEPARATOR
+            + spanContext.getSpanId().toLowerBase16()
+            + SEPARATOR
+            + DEPRECATED_PARENT_SPAN
+            + SEPARATOR
+            + sampled);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, Getter<C> getter) {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+
+    SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
+
+    return TracingContextUtils.withSpan(DefaultSpan.create(spanContext), context);
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    String value = getter.get(carrier, TRACE_ID_HEADER);
+    if (StringUtils.isNullOrEmpty(value)) {
+      return SpanContext.getInvalid();
+    }
+
+    try {
+      value = URLDecoder.decode(value, "UTF-8");
+    } catch (UnsupportedEncodingException e) {
+      logger.info(
+          "Error decoding '"
+              + TRACE_ID_HEADER
+              + "' with value "
+              + value
+              + ". Returning INVALID span context.");
+      return SpanContext.getInvalid();
+    }
+
+    String[] parts = value.split(SEPARATOR);
```

Comment:
You can avoid using split. The first separator is the only one which can be on different position then all the others are at known positions relative to the first one.

Reply:
Or the base16 encoding in the spec actually is hex encoded which means is variable size?

Reply:
If I understood the propagation spec correctly, the 3rd part (parent span id) may be set, or could be 0.  I guess it depends on which Jaeger client is being used to generate the propagation value which we wont know. 

I could look at using alternatives to ```String.split``` if we want to optimise.  Possibly ```StringTokenizer```?

Reply:
scratch the bit about ```StringTokenizer``` - just looked at it and its use in new code is discouraged.